src/proof/drat/drat_proof.cpp

```diff
@@ -0,0 +1,243 @@
+/*********************                                                        */
+/*! \file drat_proof.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Alex Ozdemir
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief DRAT Proof Format
+ **
+ ** Defines deserialization for DRAT proofs.
+ **/
+
+#include "proof/drat/drat_proof.h"
+#include <bitset>
+
+namespace CVC4 {
+namespace proof {
+namespace drat {
+
+// helper functions for parsing the binary DRAT format.
+
+/**
+ * Parses a binary literal which starts at `start` and must not go beyond `end`
+ *
+ * Leaves the iterator one past the last byte that is a part of the clause.
+ *
+ * If the literal overruns `end`, then raises a `InvalidDratProofException`.
+ */
+SatLiteral parse_binary_literal(std::string::const_iterator& start,
+                                const std::string::const_iterator& proof_end)
+{
+  // lit is encoded as uint represented by a variable-length byte sequence
+  uint64_t literal_represented_as_uint = 0;
+  for (int shift = 0; start != proof_end; ++start, shift += 7)
+  {
+    unsigned char byte = *start;
+    // The MSB of the byte is an indicator of whether the sequence continues
+    bool continued = (byte >> 7) & 1;
+    unsigned char numeric_part = byte & 0x7f;
+    literal_represented_as_uint |= numeric_part << shift;
```

Comment:
Good catch!

Agreed, I'll raise an error if the literal is too big.

I think you're correct regarding the conversion to `unsigned char`, per 
> Otherwise, both operands are converted to the unsigned counterpart of the signed operand's type. 

And yes, that could result in undefined behavior. I'll make `numeric_part` a `uint_64t` and make `shift` a `uint_8t`.